### PR TITLE
feat: added report skill for users

### LINKED ISSUE #1530: Add feature to report a skill
**Actual Behaviour**

Currently, there is no option to allow the user to report a skill.

**Expected Behaviour**

Add this option in the skill details screen.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #1530 : Feature to report a skill added!

Changes: Added the api call to report a skill

Screenshots for the change: 

screenshot-1534104032819

Edit 1: Moved the report feature to end, as similar to google assistant
screenshot-1534170024081

@arundhati24 @batbrain7 Please check!

### CODE REVIEW

**Comment by arundhati24** [comment]
Also, CI tests are failing. Please have a look and fix the issues.

**Comment by codedsun** [comment]
@batbrain7 The previous PR was not failing which was closed, and I have last night just pushed the same code as that can't be reopened. I am working on it now

**Comment by codedsun** [comment]
@batbrain7 @arundhati24 Please review!

**Comment by codedsun** [comment]
@iamareebjamal @arundhati24 @batbrain7 Review plz

**Comment by codedsun** [comment]
@batbrain7  @arundhati24 Please check!

**Comment by arundhati24** [comment]
@codedsun Please update the screenshots.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 1]
```diff
@@ -118,6 +121,40 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
...
     }
 
+    private fun setReportButton() {
+        //Only enable the feature when the user is logged in
+        if (PrefManager.getToken() != null) {
+            reportSkill.visibility = View.VISIBLE
+            reportSkill.setOnClickListener {
+                val dialogBuilder = AlertDialog.Builder(requireContext())
+                val view = layoutInflater.inflate(R.layout.alert_report_skill, null)
+                val reportMsgText = view.findViewById(R.id.reportMsg) as EditText
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Use a proper variable name. Avoid short notations like 'Msg'

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 2]
```diff
@@ -118,6 +121,40 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
...
+
+                dialogBuilder.setPositiveButton(R.string.report_send) { dialog, whichButton ->
+                    if (PrefManager.getToken() != null && reportMsgText.text.isNotEmpty()) {
+                        val map: MutableMap<String, String> = HashMap()
+                        map.put("access_token", PrefManager.getToken().toString())
+                        map.put("feedback", reportMsgText.text.toString())
+                        map.put("skill", skillTag)
+                        map.put("group", skillData.group)
+                        map.put("model", skillData.model)
+                        skillDetailsPresenter.sendReport(map)
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Use objects.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt [hunk 3]
```diff
@@ -137,6 +142,19 @@ class SkillDetailsPresenter(skillDetailsFragment: SkillDetailsFragment) : ISkill
         }
     }
 
+    override fun reportSendError(throwable: Throwable) {
+        Timber.d(throwable)
+        skillDetailsView?.updateSkillReportStatus("Coudn't report the skill")
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: Use string resource.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt [hunk 4]
```diff
@@ -137,6 +142,19 @@ class SkillDetailsPresenter(skillDetailsFragment: SkillDetailsFragment) : ISkill
...
+    }
+
+    override fun reportSendSuccess(response: Response<ReportSkillResponse>) {
+        if (response.code() == 422) {
+            skillDetailsView?.updateSkillReportStatus("Skill Already Reported")
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: Use string resource. Why capital 'A' and 'R'?

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt [hunk 5]
```diff
@@ -137,6 +142,19 @@ class SkillDetailsPresenter(skillDetailsFragment: SkillDetailsFragment) : ISkill
...
+        } else {
+            skillDetailsView?.updateSkillReportStatus("Reported Successfully")
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: same

File: app/src/main/res/layout/alert_report_skill.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:paddingLeft="@dimen/activity_vertical_margin"
+    android:paddingRight="@dimen/activity_vertical_margin"
+    android:paddingTop="@dimen/margin_large">
+
+    <EditText
+        android:id="@+id/reportMsg"
```
Comment (arundhati24) [inline app/src/main/res/layout/alert_report_skill.xml]: use proper id name, avoid short notations like 'Msg'

File: app/src/main/res/layout/alert_report_skill.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:paddingLeft="@dimen/activity_vertical_margin"
+    android:paddingRight="@dimen/activity_vertical_margin"
+    android:paddingTop="@dimen/margin_large">
```
Comment (arundhati24) [inline app/src/main/res/layout/alert_report_skill.xml]: @dimen/padding_large

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 8]
```diff
@@ -47,6 +47,7 @@
                     android:layout_marginTop="8dp"
                     android:background="@color/md_white_1000"
                     android:text="@string/author_name" />
+
```
Comment (arundhati24) [inline app/src/main/res/layout/fragment_skill_details.xml]: unrelated change

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 9]
```diff
@@ -23,7 +23,7 @@
                 app:srcCompat="@drawable/ic_susi" />
 
             <LinearLayout
-                android:layout_width="wrap_content"
+                android:layout_width="match_parent"
```
Comment (arundhati24) [inline app/src/main/res/layout/fragment_skill_details.xml]: What was the need of this?

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt (hunk 3 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: @arundhati24 How can i get the context, it's presenter class.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 2 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: @arundhati24 Couldn't get it?Please tell me more

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt (hunk 4 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: same

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt (hunk 3 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: You can check SkillListingPresenter for reference.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 10]
```diff
@@ -118,6 +119,40 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
...
+
+                dialogBuilder.setPositiveButton(R.string.report_send) { dialog, whichButton ->
+                    if (PrefManager.getToken() != null && reportMessage.text.isNotEmpty()) {
+                        val map: MutableMap<String, String> = HashMap()
+                        map.put("access_token", PrefManager.getToken().toString())
+                        map.put("feedback", reportMessage.text.toString())
+                        map.put("skill", skillTag)
+                        map.put("group", skillData.group)
+                        map.put("model", skillData.model)
+                        skillDetailsPresenter.sendReport(map)
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: use objects. You may use a data class for the same. Pass the object rather than passing map as param. Then, pass map to the final method, defined in SusiService.java, using ClientBuilder.java. Please check the implementation for other features like feedback for a reference.

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -399,4 +399,11 @@
     <string name="metric_newest">\"SUSI, what are the newest skills?\"</string>
     <string name="metrics_top_games">\"SUSI, what are your top games?\"</string>
     <string name="message_no_skills_found">No skills found.</string>
+
+    <string name="report_skill">Report Skill</string>
+    <string name="report_send">Send Report</string>
+    <string name="report_error">Coudn\'t report the skill</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: 'Error reporting skill' or may be check some other standard apps for a more standard string.

File: app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -399,4 +399,11 @@
...
+    <string name="skill_already_reported">Skill already reported</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: please use proper naming conventions. In most of the cases, you have used the message as the key name. You could say 'message_report_skill', 'error_report_skill' etc. Please check it for other added strings as well.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 13]
```diff
@@ -118,6 +119,40 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
...
     }
 
+    private fun setReportButton() {
+        //Only enable the feature when the user is logged in
+        if (PrefManager.getToken() != null) {
+            reportSkill.visibility = View.VISIBLE
+            reportSkill.setOnClickListener {
+                val dialogBuilder = AlertDialog.Builder(requireContext())
+                val view = layoutInflater.inflate(R.layout.alert_report_skill, null)
+                val reportMessage = view.findViewById(R.id.reportMessage) as EditText
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Please check if ```reportMessage``` can be referenced directly in the Kotlin file.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 10 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: @arundhati24 What effect will it have? I have just implemented the same way all other functionalities are implemented.

Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: @arundhati24 In the implementation of feedback, we have already extracted the same things, then why are we sending another object and do the same thing again?

In SkillDetailsModel.

```
 val query: MutableMap<String, String> = HashMap()
        query.put("model", queryObject.model)
        query.put("group", queryObject.group)
        query.put("language", queryObject.language)
        query.put("skill", queryObject.skill)
        query.put("feedback", queryObject.feedback)
        query.put("access_token", queryObject.accessToken)
```

same fields again extracted.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 13 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: @arundhati24 we have to inflate the xml and then reference from that view, that's why we have to apply findViewById.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 10 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: You can refer to #1491 and also PRs related to feedback features as this has been discussed a lot already. We don't want to use maps directly as it is not safe. Also, as our mentor  @iamareebjamal has already mentioned - not to use maps in views.

**Comment by arundhati24** [comment]
Also, please have a look at the issues, other than the documentation issues, listed by Codacy.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 10 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: @arundhati24 Sure, moving the code and sending the PR asap.!

**Comment by codedsun** [comment]
@arundhati24 Check now!

File: app/src/main/java/org/fossasia/susi/ai/data/SkillDetailsModel.kt [hunk 14]
```diff
@@ -112,6 +115,27 @@ class SkillDetailsModel : ISkillDetailsModel {
...
     }
 
+    override fun sendReport(queryObject: ReportSkill, listener: ISkillDetailsModel.OnReportSendListener) {
+        val map: MutableMap<String, String> = HashMap()
+        map.put("access_token", queryObject.accessToken)
+        map.put("feedback", queryObject.feedback)
+        map.put("skill", queryObject.skill)
+        map.put("group", queryObject.group)
+        map.put("model", queryObject.model)
+        reportSkillResponseCall = ClientBuilder.getSusiApi().reportSkill(map)
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/data/SkillDetailsModel.kt]: Add a static method in ClientBuilder.java and pass an object to that method from here. In the defined method, create a map and return a ```Call<ReportSkillResponse>```.

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/ReportSkillResponse.kt [hunk 15]
```diff
@@ -0,0 +1,13 @@
+package org.fossasia.susi.ai.rest.responses.susi
+
+/**
+ * @author : codedsun
+ * Created on 06/08/18
+ */
+/**
+ * Response received on successful report of skill
+ */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/ReportSkillResponse.kt]: This comment can be added to the first comment, above author name separated by a line, IMO.

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -402,4 +402,11 @@
     <string name="metric_newest">\"SUSI, what are the newest skills?\"</string>
     <string name="metrics_top_games">\"SUSI, what are your top games?\"</string>
     <string name="message_no_skills_found">No skills found.</string>
+
+    <string name="report_skill">Report Skill</string>
+    <string name="report_send_msg">Send Report</string>
+    <string name="report_error_msg">Coudn\'t report the skill</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: ```An error occurred. Please try again later.```

**Comment by codedsun** [comment]
@arundhati24 The screenshots have been added for the changes earlier in the issue description section.

**Comment by arundhati24** [comment]
@codedsun Yes, that's right. But, as there have been a lot of UI changes recently, hence, I request you to add an updated screenshot. That would help me have a clear idea of the changes that you have made before approving this PR.
Also, please check the text size, color and other things as per the latest UI changes so as to avoid another set of reviews and follow up PRs.

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/ReportSkillResponse.kt [hunk 17]
```diff
@@ -0,0 +1,15 @@
+package org.fossasia.susi.ai.rest.responses.susi
+
+/**
+ * Response received on successful report of skill
+ */
+
+/**
+ * @author : codedsun
+ * Created on 06/08/18
+ */
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/ReportSkillResponse.kt]:
```
/**
 * Response received on reporting a skill successfully
 *
 * @author : codedsun
 * Created on 06/08/18
 */
```

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt [hunk 18]
```diff
@@ -2,10 +2,12 @@ package org.fossasia.susi.ai.data.contract
 
 import org.fossasia.susi.ai.dataclasses.FetchFeedbackQuery
 import org.fossasia.susi.ai.dataclasses.PostFeedback
+import org.fossasia.susi.ai.dataclasses.ReportSkill
 import org.fossasia.susi.ai.rest.responses.susi.FiveStarSkillRatingResponse
 import org.fossasia.susi.ai.rest.responses.susi.GetRatingByUserResponse
-import org.fossasia.susi.ai.rest.responses.susi.GetSkillFeedbackResponse
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt]: Unrelated change?

File: app/src/main/java/org/fossasia/susi/ai/dataclasses/ReportSkill.kt [hunk 19]
```diff
@@ -0,0 +1,14 @@
+package org.fossasia.susi.ai.dataclasses
+
+/**
+ * @author : codedsun
+ * Created on 25/08/18
+ */
+
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/dataclasses/ReportSkill.kt]: Remove line.

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/ReportSkillResponse.kt [hunk 20]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/ReportSkillResponse.kt]: Remove this extra line.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 21]
```diff
@@ -26,17 +28,16 @@ import org.fossasia.susi.ai.R
 import org.fossasia.susi.ai.chat.ChatActivity
 import org.fossasia.susi.ai.dataclasses.FetchFeedbackQuery
 import org.fossasia.susi.ai.dataclasses.PostFeedback
+import org.fossasia.susi.ai.dataclasses.ReportSkill
 import org.fossasia.susi.ai.helper.PrefManager
 import org.fossasia.susi.ai.helper.Utils
 import org.fossasia.susi.ai.rest.responses.susi.GetSkillFeedbackResponse
 import org.fossasia.susi.ai.rest.responses.susi.SkillData
 import org.fossasia.susi.ai.rest.responses.susi.Stars
-import org.fossasia.susi.ai.skills.SkillsActivity
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Unrelated change.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 22]
```diff
@@ -26,17 +28,16 @@ import org.fossasia.susi.ai.R
...
 import org.fossasia.susi.ai.skills.skilldetails.adapters.recycleradapters.FeedbackAdapter
 import org.fossasia.susi.ai.skills.skilldetails.adapters.recycleradapters.SkillExamplesAdapter
 import org.fossasia.susi.ai.skills.skilldetails.contract.ISkillDetailsPresenter
 import org.fossasia.susi.ai.skills.skilldetails.contract.ISkillDetailsView
-import java.io.Serializable
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Unrelated change.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt [hunk 23]
```diff
@@ -137,6 +146,19 @@ class SkillDetailsPresenter(skillDetailsFragment: SkillDetailsFragment) : ISkill
         }
     }
 
+    override fun reportSendError(throwable: Throwable) {
+        Timber.d(throwable)
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsPresenter.kt]: Timber.e

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 22 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: These are unused imports in dev branch

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 21 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: These are unused imports in dev branch

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt (hunk 18 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt]: Added on line 9

**Comment by codedsun** [comment]
screenshot-1535190343855

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 22 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Please do not remove these here. Open an issue for the same, and remove unused imports from all the files, in one PR.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt (hunk 21 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: Please do not remove these here. Open an issue for the same, and remove unused imports from all the files, in one PR.

**Comment by arundhati24** [comment]
The background of the Report Skill is '@color/default_bg'. Change it to '@color/md_white_1000'. Also, check the font family, text size and text color from fragment_skill_details.xml. It should be same as other views.

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt (hunk 18 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt]: IMO please let it be where it was.

**Comment by arundhati24** [comment]
@codedsun Status of this PR?

**Comment by arundhati24** [comment]
@codedsun It has been 16 days since you opened this PR. I request you to please finalise this now. I have tried to very clearly mention the changes needed. Please consider those changes and please finalise this now.

**Comment by codedsun** [comment]
@arundhati24 Done the changes!

**Comment by iamareebjamal** [comment]
Please follow commit guidelines https://chris.beams.io/posts/git-commit/

**Comment by codedsun** [comment]
@iamareebjamal Stuck in changing the previous commit message, it was a very OLD PR hence the commit mssage was wrong.

**Comment by codedsun** [comment]
@iamareebjamal I was performing git rebase and then changed the message successfully, but that on pushing to remote repo all the commits under one single PR.

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt [hunk 24]
```diff
@@ -2,10 +2,12 @@ package org.fossasia.susi.ai.data.contract
 
 import org.fossasia.susi.ai.dataclasses.FetchFeedbackQuery
 import org.fossasia.susi.ai.dataclasses.PostFeedback
+import org.fossasia.susi.ai.dataclasses.ReportSkill
 import org.fossasia.susi.ai.rest.responses.susi.FiveStarSkillRatingResponse
-import org.fossasia.susi.ai.rest.responses.susi.GetRatingByUserResponse
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt]: unrelated change. Please do not make this change here. Let it be on line 6.

File: app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt [hunk 25]
```diff
@@ -26,17 +28,18 @@ import org.fossasia.susi.ai.R
...
 import org.fossasia.susi.ai.rest.responses.susi.GetSkillFeedbackResponse
 import org.fossasia.susi.ai.rest.responses.susi.SkillData
 import org.fossasia.susi.ai.rest.responses.susi.Stars
-import org.fossasia.susi.ai.skills.SkillsActivity
 import org.fossasia.susi.ai.skills.skilldetails.adapters.recycleradapters.FeedbackAdapter
+import java.io.Serializable
 import org.fossasia.susi.ai.skills.skilldetails.adapters.recycleradapters.SkillExamplesAdapter
 import org.fossasia.susi.ai.skills.skilldetails.contract.ISkillDetailsPresenter
 import org.fossasia.susi.ai.skills.skilldetails.contract.ISkillDetailsView
-import java.io.Serializable
```
Comment (arundhati24) [inline app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt]: @codedsun I reviewed this last time. How many times should I ask you for the same change? Please please consider all reviews and make the changes properly and carefully. I told you not to remove these here in this PR.
This is the reason why this PR is still pending. :(

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 26]
```diff
@@ -477,6 +477,16 @@
...
                 app:fontFamily="sans-serif" />
 
+            <TextView
+                android:id="@+id/reportSkill"
+                android:visibility="gone"
+                android:background="@color/md_white_1000"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:textColor="@color/md_grey_700"
+                android:textSize="16sp"
```
Comment (arundhati24) [inline app/src/main/res/layout/fragment_skill_details.xml]: Hardcoded dimen.

File: app/src/main/res/values/strings.xml [hunk 27]
```diff
@@ -402,4 +402,11 @@
     <string name="metric_newest">\"SUSI, what are the newest skills?\"</string>
     <string name="metrics_top_games">\"SUSI, what are your top games?\"</string>
     <string name="message_no_skills_found">No skills found.</string>
+
+    <string name="report_skill">Report Skill</string>
+    <string name="report_send_msg">Send Report</string>
+    <string name="report_error_msg">An error occured. Please try again later</string>
+    <string name="skill_already_reported">Skill already reported</string>
+    <string name="report_success_msg">Reported successfully</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: Skill reported successfully

**Comment by codedsun** [comment]
@arundhati24 There are many issues which I am facing due to changing the last commit message. And I think I should send a fresh PR for both same, see the issue here https://github.com/fossasia/susi_android/pull/1608

**Comment by arundhati24** [comment]
@codedsun What kind of issues are you facing? Did you try to find out how to fix those issues? 
And, if you have to send a new PR, please make sure that you incorporate all requested changes and all reviews. It's difficult to ask for same changes n number of times.

**Comment by codedsun** [comment]
Closing this PR! Pushed a fresh one
